netbox/dcim/models/device_components.py

```diff
@@ -97,7 +98,7 @@ def to_objectchange(self, action):
     def clean(self):
         super().clean()
 
-        if self.pk is not None and self._original_device != self.device_id:
+        if (not self._can_switch_device) and (self.pk is not None) and (self._original_device != self.device_id):
```

Comment:
I'm not sure setting a variable to indicate whether a component is moveable is the ideal pattern. Instead, we could define a list of exempt classes:

```suggestion
        if type(self) not in [InventoryItem] and self.pk is not None and self._original_device != self.device_id:
```